### ISSUE TITLE
ColorPicker: use modern JS widget

### ISSUE THREAD

**Issue by jepsar** [primefaces/primefaces#8000]
As the current 3rd party code has not been touched in 10 years and contains at least this bug:  https://github.com/primefaces/primefaces/issues/7997

**Options:**
Coloris: https://github.com/melloware/Coloris
IRO: https://iro.js.org/colorPicker_api.html

**Comment by melloware** [comment]
I think the current PrimeNG widget doesn't offer all the functionality of our legacy component.  I am all for a new component but it should replace the functionality of the current one. Coloris looks great but it doesn't look like it goes RGB or HSB values which our current component does.

**Comment by jepsar** [comment]
Right, HSL is a must have. I'll open a ticket there.

**Comment by melloware** [comment]
Sorry Coloris does do RGB!  checking if it does HSB.

**Comment by jepsar** [comment]
It does now.

**Comment by jepsar** [comment]
https://iro.js.org/colorPicker_api.html

**Comment by jepsar** [comment]
I'll go for https://iro.js.org/colorPicker_api.html which is way more flexible and allows us to replace the current picker in PF without breaking changes and add features later (like alpha, hsla, rgba, wheel, swatches etc).

**Comment by melloware** [comment]
I like that idea.

**Comment by jepsar** [comment]
Okay, so I've spent some time on trying to replace the jQuery color picker with iro.js. Inline is easy, but there is no popup support. I though that popup was added by our widget, but I found out that most of it is part of the jQuery plugin. Dropping popup is not really an option, and adding popup support will take too much time IMO, especially when we know that all components will be made accessible soon. Since both the current jQuery plugin and iro.js are not accessible, a replacement based on the React code is expected.

See also: https://github.com/jaames/iro.js/issues/237, https://github.com/jaames/iro.js/issues/238 and https://github.com/jaames/iro.js/pull/229

**Comment by melloware** [comment]
I wonder if we can make Coloris work if we hack the component to have multiple versions? It has all the features we need I believe with accessibility support I think I just need to hack the component to allow multiple instances?  

But you are right maybe waiting to see how the PrimeReact component comes out first might be the right move.

**Comment by jepsar** [comment]
Yeah, I was hoping for an easy swap.
